docs/observability/aws/deploy-use-aws-observability/migration-strategy-using-terraform.md

```diff
@@ -0,0 +1,110 @@
+    sumologic_collector_details = {
+        "collector_name": "AWS Observability Prod",
+        "description": "This collector is created using Sumo Logic terraform AWS Observability module.",
+        "fields": {}
+    }
+    collect_cloudwatch_logs = "Kinesis Firehose Log Source"
+    collect_cloudwatch_metrics = "Kinesis Firehose Metrics Source"
+    collect_cloudtrail_logs = "false"
+    collect_classic_lb_logs = "false"
+    collect_elb_logs = "false"
+    collect_root_cause_data = "None"
+    }
+    ```
+1. Verify the following:
+    1. Log group and metrics subscription. 
+    1. Logs and metrics ingested into Sumo Logic via Kinesis Firehose sources.
+1. Once verified, delete the old CloudWatch log sources (HTTP) and metrics sources with other associated resources like Lambdas, subscription, and more.
+    1. Disable logs and metrics collection in the original collection block and run `terraform apply` again to clean up the redundant resources.
+    ```sql
+    module "old-collection-module" {
+    source = "./source-module"
+
+    aws_account_alias         = var.aws_account_alias
+    sumologic_organization_id = var.sumologic_organization_id
+    access_id                 = var.sumologic_access_id
+    access_key                = var.sumologic_access_key
+    environment               = var.sumologic_environment  
+    collect_cloudwatch_logs = "None"
+    collect_cloudwatch_metrics = "None"
+    }
+    ```
+
+:::note
+This approach will create an additional collector with Kinesis logs and metrics sources.
+:::
+
+### Advantages
+
+- There is no loss of logs and metrics during migration.
+
+### Disadvantages
+
+- There will be ingestion of duplicate logs and metrics.
+
+## Approach 2: Update existing collection module
+
+1. Update `collect_cloudwatch_logs` from **Lambda Log Forwarder** to **Kinesis Firehose Log Source**.
+1. Update `collect_cloudwatch_metrics` from **CloudWatch Metrics Source** to **Kinesis Firehose Metrics Source**.
+    ```sql
+    collect_cloudwatch_logs = "Kinesis Firehose Log Source"
+    collect_cloudwatch_metrics = "Kinesis Firehose Metrics Source"
+    ```
+1. Run `terraform apply`.
+    - Forwarding of logs and metrics will start to Sumo Logic using Kinesis Firehose for Logs and Kinesis Firehose for Metrics.
+    - Old CloudWatch log source (HTTP) and metrics source along with other associated resources such as Lambdas, subscription, and more will also be deleted.
+
+### Advantages
+
+- There is no ingestion of duplicate logs and metrics.
+
+### Disadvantages
+
+- There will be a loss of logs and metrics during migration.
+
+## Troubleshooting
+
+### Error
+
+`A resource with the same name already exists`
+
+### Cause
+
+When you try to migrate CloudWatch Source to Kinesis Firehose Source using [Approach 2](#approach-2-update-existing-collection-module), migration will fail because Terraform tries to create a Kinesis source with the same name before deleting the existing Cloudwatch Source. 
```

Comment:
```suggestion
When you try to migrate CloudWatch Source to Kinesis Firehose Source using [Approach 2](#approach-2-update-existing-collection-module), migration will fail. This is because Terraform tries to create a Kinesis source with the same name before deleting the existing CloudWatch Source. 
```